notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/NetworkStatusReceiver.java

```diff
@@ -0,0 +1,40 @@
+package com.microsoft.windowsazure.messaging.notificationhubs;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+/**
+ * Synchronizes updates to the tags or other installation aware items on the device
+ * when the device comes back online.
+ */
+public class NetworkStatusReceiver extends BroadcastReceiver {
+
+    private final NotificationHub mNotificationHub;
+
+    public NetworkStatusReceiver() {
+        mNotificationHub = NotificationHub.getInstance();
```

Comment:
Why cache the singleton instance, versus just calling the static method? This is inconsistent with other similar scenarios (like `NotificationHub.relayMessage()`).

Reply:
Basically to allow for better testing. I was thinking of actually fixing the inconsistency of this by using this pattern in all the other similar scenarios. Thoughts?

Reply:
Fixed in e14bec3b86ebc591db44ca86ef1608dc917ae2fa